crypto/x509/by_dir.c

```diff
@@ -420,9 +421,10 @@ static int get_cert_by_subject_ex(X509_LOOKUP *xl, X509_LOOKUP_TYPE type,
  finish:
     /* If we changed anything, resort the objects for faster lookup */
     if (!sk_X509_OBJECT_is_sorted(xl->store_ctx->objs)) {
-        X509_STORE_lock(xl->store_ctx);
-        sk_X509_OBJECT_sort(xl->store_ctx->objs);
-        X509_STORE_unlock(xl->store_ctx);
+        if(X509_STORE_lock(xl->store_ctx)) {
```

Comment:
(minor, whitespace) coding standard would like a space after `if`

Reply:
Thanks for your review. I fixed it